src/psyclone/psyir/backend/sympy_writer.py

```diff
@@ -343,3 +508,147 @@ def is_intrinsic(self, operator):
+        each array index into a three parameters to support array expressions.
+
+        :param indices: list of PSyIR nodes.
+        :type indices: List[:py:class:`psyclone.psyir.symbols.Node`]
+        :param str var_name: name of the variable for which the dimensions \
+            are created. Not used in the Fortran implementation.
+
+        :returns: the Fortran representation of the dimensions.
+        :rtype: List[str]
+
+        :raises NotImplementedError: if the format of the dimension is not \
+            supported.
+
+        '''
+        dims = []
+        for index in indices:
+            if isinstance(index, DataNode):
+                # literal constant, symbol reference, or computed
+                # dimension
+                expression = self._visit(index)
+                dims.extend([expression, expression, "1"])
+            elif isinstance(index, Range):
+                # literal constant, symbol reference, or computed
+                # dimension
+                expression = self._visit(index)
+                dims.append(expression)
+            elif isinstance(index, ArrayType.ArrayBounds):
+                # Lower and upper bounds of an array declaration specified
+                # by literal constant, symbol reference, or computed dimension
+                lower_expression = self._visit(index.lower)
+                upper_expression = self._visit(index.upper)
+                dims.extend([lower_expression, upper_expression, "1"])
+            elif isinstance(index, ArrayType.Extent):
+                # unknown extent
+                dims.extend([self.lower_bound_name, self.upper_bound_name,
+                            "1"])
+            else:
+                raise NotImplementedError(
+                    f"unsupported gen_indices index '{index}'")
+        return dims
+
+    # -------------------------------------------------------------------------
+    def range_node(self, node):
+        '''This method is called when a Range instance is found in the PSyIR
+        tree. This implementation converts a range into three parameters
+        for the corresponding SymPy function.
+
+        :param node: a Range PSyIR node.
+        :type node: :py:class:`psyclone.psyir.nodes.Range`
+
+        :returns: the Fortran code as a string.
+        :rtype: str
+
+        '''
+        if node.parent and node.parent.is_lower_bound(
+                node.parent.indices.index(node)):
+            # The range starts for the first element in this
+            # dimension. This is the default in Fortran so no need to
+            # output anything.
+            start = self.lower_bound_name
+        else:
+            start = self._visit(node.start)
+
+        if node.parent and node.parent.is_upper_bound(
+                node.parent.indices.index(node)):
+            # The range ends with the last element in this
+            # dimension. This is the default in Fortran so no need to
+            # output anything.
```

Comment:
cut-n-pasted comment?

Reply:
And still C&P

Reply:
I bet CodePilot doesn't have this problem :-)